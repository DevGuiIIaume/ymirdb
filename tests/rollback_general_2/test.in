set a 1
set b a 2
set c 3
snapshot
set d c 4
snapshot
list entries
del b
del d
del a
del c
list entries
checkout 1
list entries
del c
del b
del a
list entries
checkout 2
list entries
del d
del b
del a
del c
list entries
checkout 2
list entries
rollback 1
list entries
snapshot
del c
del b
del a
list entries
checkout 3
list entries
checkout 1
list entries
forward a
forward b
backward a
forward c
backward b
backward c
checkout 3
forward a
forward b
forward c
backward a
backward b
backward c
bye
