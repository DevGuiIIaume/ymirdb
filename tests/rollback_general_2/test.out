> ok

> ok

> ok

> saved as snapshot 1

> ok

> saved as snapshot 2

> d [c 4]
c [3]
b [a 2]
a [1]

> ok

> ok

> ok

> ok

> no entries

> ok

> c [3]
b [a 2]
a [1]

> ok

> ok

> ok

> no entries

> ok

> d [c 4]
c [3]
b [a 2]
a [1]

> ok

> ok

> ok

> ok

> no entries

> ok

> d [c 4]
c [3]
b [a 2]
a [1]

> ok

> c [3]
b [a 2]
a [1]

> saved as snapshot 3

> ok

> ok

> ok

> no entries

> ok

> c [3]
b [a 2]
a [1]

> ok

> c [3]
b [a 2]
a [1]

> nil

> a

> b

> nil

> nil

> nil

> ok

> nil

> a

> nil

> b

> nil

> nil

> bye
